lib/migrations/createGenesisDao.ts

```diff
@@ -23,135 +29,130 @@ interface FounderSpec {
+    const Avatar = await Utils.requireContract("Avatar");
+
+    const DaoCreator = await Utils.requireContract("DaoCreator");
+    const daoCreatorInst = await DaoCreator.deployed();
+    const UController = await Utils.requireContract("UController");
+    const universalControllerInst = await UController.deployed();
 
-    console.log(`Deploying to ${network}, gasLimit: ${gasAmount}, ${founders.length} founders`);
+    const internalFoundersConfigLocation = "../../migrations/founders.json";
+    const foundersConfig = require(internalFoundersConfigLocation).founders;
 
-    await deployer.deploy(ControllerCreator, { gas: gasAmount });
-    const controllerCreator = await ControllerCreator.deployed();
-    await deployer.deploy(DaoCreator, controllerCreator.address, { gas: gasAmount });
-    const daoCreatorInst = await DaoCreator.deployed(controllerCreator.address);
+    const customFoundersConfigLocation = foundersConfigurationLocation || internalFoundersConfigLocation;
+
+    if (internalFoundersConfigLocation !== customFoundersConfigLocation) {
+      console.log(`merging custom founders from ${customFoundersConfigLocation}`);
+      const customFoundersConfig = require(customFoundersConfigLocation).founders;
+      // merge the two
+      Object.assign(foundersConfig, customFoundersConfig);
+    }
+
+    const founders = foundersConfig[this.network];
+
+    if (!founders || (founders.length === 0)) {
+      throw new Error(`no founders were given for the network: ${this.network}`);
+    }
+
+    let gasLimit = computeForgeOrgGasLimit(founders.length);
+    const maxGasLimit = await computeMaxGasLimit(this.web3);
+
+    gasLimit = Math.min(gasLimit, maxGasLimit);
+
+    console.log(`Forging ${orgName} to ${this.network}, gasLimit: ${gasLimit} and ${founders.length} founders...`);
 
-    await deployer.deploy(UController, { gas: gasAmount });
-    const universalControllerInst = await UController.deployed();
     /**
      * Create the Genesis DAO
      */
-    const tx = await daoCreatorInst.forgeOrg(
+    const txForgeOrg = await daoCreatorInst.forgeOrg(
       orgName,
       tokenName,
       tokenSymbol,
       founders.map((f: FounderSpec) => f.address),
-      founders.map((f: FounderSpec) => web3.toWei(f.tokens)),
-      founders.map((f: FounderSpec) => web3.toWei(f.reputation)),
+      founders.map((f: FounderSpec) => this.web3.toWei(f.tokens)),
+      founders.map((f: FounderSpec) => this.web3.toWei(f.reputation)),
       universalControllerInst.address,
-      web3.toWei(100000000), // token cap of one hundred million GEN, in Wei
-      { gas: gasAmount });
+      this.web3.toWei(100000000), // token cap of one hundred million GEN, in Wei
+      { gas: gasLimit });
 
-    // maximize chances that Avatar.at will succeed
-    if (network === "live") {
-      await UtilsInternal.sleep(60000);
+    /**
+     * save info for later steps
+     */
+    if (live) {
+      UtilsInternal.sleep(10000); // maximize chances this is ready
```

Comment:
Does this sleep is a must ?
I am not sure.
We can use a better way to confirm all is ready such as checking it..

Reply:
Refactored to only sleep when necessary

Reply:
If the script written in a good way it should not use  sleep at all...

Reply:
We still don't know why it was needed in the first place. Until we understand, I would prefer to leave it in.  I have commented the code accordingly.